config.go

```diff
@@ -7,9 +7,89 @@ import (
 	"time"
 )
 
+// These are the versions of the protocol (which includes RPC messages as
+// well as Raft-specific log entries) that this server can _understand_. Use
+// the ProtocolVersion member of the Config object to control the version of
+// the protocol to use when _speaking_ to other servers. This is not currently
+// written into snapshots so they are unversioned. Note that depending on the
```

Comment:
Er re-reading it again I think it's safe _at startup_ time because whoever becomes leader won't ever change their log and will enforce their log on the followers, and in this case it will have to happen with an install snapshot after a recovery.

I still like the change on the Consul side, though, because there really is no reason to run recovery on first startup. Going to sleep now!